### PR TITLE
Users most recent input location can now be recorded in the config file.

### PR DESCRIPTION
**Pull request by ModoUnreal**
So this feature has finally come to fruition. This basically makes another section in the config file, this being the HISTORY section, where 5 variables holding recently searched locations are saved. So the most recent location is added to recent_1 and all other variables go down to recent_2 and so on.

Test this out to see whether it works.
If there is anything else missing with this feature then please inform me of it, I will try to do it as quick as possible.

### CODE REVIEW

**Comment by o355** [comment]
Thanks for the PR, I'll have a look at it when I have the time.

**Comment by o355** [comment]
Hey, thanks for the PR. Sadly, this isn't what was planned to run in PyWeather, but you've got the base code.

You'll need to implement showing the user their previous locations at boot, as this is what was intended. Code to show you this can be found already in PyWeather, so it shouldn't be that hard to work in. Add one configuration option for showing previous locations at boot.

You'll also need code to make sure `currentlocation` or `curloc` queries are not counted for the current location, and that you properly parse PWS queries. For PWS queries, keep the raw `pws:PWSGOESHERE` in the config file, and what gets entered into the `locinput` field. When displaying to the user, display `previouslocation:1 - PWS PWSGOESHERE`.

Basically, you'll need 10 variables for previous locations. The raw location used to query into `locinput`, and the display variables. In the preload, you'll have to define a section to run an if loop, and set display variables. For PWSes, do this by removing `pws:` from the raw location, and `.upper()` to get the PWS ID. After that, add `PWS `, and you've got your display variable for the previous location.

If you're confused, I'm fine with coding it in myself. You can work on adding hurricane cache times and yesterday's weather cache times and prefetching at boot if you feel this is too hard.

I'm going to close this as I don't want this to end up the way that yesterday's weather did.

**Comment by ModoUnreal** [comment]
I'll try to work on this. Really the only difficulty I will have is regarding the PWS, since it is a fairly new concept for me. Again, I will attempt and will contact you on Reddit if any difficulties arise.
